src/System.Memory/tests/ReadOnlySpan/IndexOfAny.char.cs

```diff
@@ -321,11 +321,276 @@ public static void MakeSureNoChecksGoOutOfRangeThree_Char()
             }
         }
 
+        [Fact]
+        public static void ZeroLengthIndexOfFour_Char()
+        {
+            ReadOnlySpan<char> sp = new ReadOnlySpan<char>(Array.Empty<char>());
+            ReadOnlySpan<char> values = new char[] { (char)0, (char)0, (char)0, (char)0 };
+            int idx = sp.IndexOfAny<char>(values);
+            Assert.Equal(-1, idx);
+        }
+
+        [Fact]
+        public static void DefaultFilledIndexOfFour_Char()
+        {
+            Random rnd = new Random(42);
+
+            for (int length = 0; length < byte.MaxValue; length++)
+            {
+                char[] a = new char[length];
+                ReadOnlySpan<char> span = new ReadOnlySpan<char>(a);
+
+                char[] targets = { default, (char)99, (char)98, (char)97 };
+
+                for (int i = 0; i < length; i++)
+                {
+                    int index = rnd.Next(0, 4);
+                    ReadOnlySpan<char> values = new char[] { (char)targets[index], (char)targets[(index + 1) % 2], (char)targets[(index + 1) % 3], (char)targets[(index + 1) % 4] };
+                    int idx = span.IndexOfAny(values);
+                    Assert.Equal(0, idx);
+                }
+            }
+        }
+
+        [Fact]
+        public static void TestMatchFour_Char()
+        {
+            for (int length = 0; length < byte.MaxValue; length++)
+            {
+                char[] a = new char[length];
+                for (int i = 0; i < length; i++)
+                {
+                    a[i] = (char)(i + 1);
+                }
+                ReadOnlySpan<char> span = new ReadOnlySpan<char>(a);
+
+                for (int targetIndex = 0; targetIndex < length; targetIndex++)
+                {
+                    ReadOnlySpan<char> values = new char[] { (char)a[targetIndex], (char)0, (char)0, (char)0 };
+                    int idx = span.IndexOfAny(values);
+                    Assert.Equal(targetIndex, idx);
+                }
+
+                for (int targetIndex = 0; targetIndex < length - 3; targetIndex++)
+                {
+                    ReadOnlySpan<char> values = new char[] { (char)a[targetIndex], (char)a[targetIndex + 1], (char)a[targetIndex + 2], (char)a[targetIndex + 3] };
+                    int idx = span.IndexOfAny(values);
+                    Assert.Equal(targetIndex, idx);
+                }
+
+                for (int targetIndex = 0; targetIndex < length - 3; targetIndex++)
+                {
+                    ReadOnlySpan<char> values = new char[] { (char)0, (char)0, (char)0, (char)0, (char)a[targetIndex + 3] };
```

Comment:
The test is named "match four"... this has 5.  Is that intentional?

Reply:
Nope, that's wrong

Reply:
Fixed